tuning/data/data_preprocessing_utils.py

```diff
@@ -0,0 +1,208 @@
+        json_dataset = datasets.load_dataset(
+            "json", data_files=data_args.training_data_path
+        )
+        if DEFAULT_JSON_INPUT_KEY not in json_dataset["train"].column_names:
+            raise ValueError(
+                "JSON should contain input field if no dataset_text_field or \
+                     data_formatter_template specified"
+            )
+        if DEFAULT_JSON_OUTPUT_KEY not in json_dataset["train"].column_names:
+            raise ValueError(
+                "JSON should contain output field if no dataset_text_field or \
+                    data_formatter_template specified"
+            )
+
+
+### Utils for custom masking / manipulating input / output strs, etc
+def combine_sequence(input_element: str, output_element: str, eos_token: str = ""):
+    """Combines / concatenates input & output element.
+
+    Args:
+        input_element: str
+            Input component of the combined sequence.
+        output_element: str
+            Output component of the combined sequence.
+        eos_token: str
+            EOS token associated with the tokenizer. \
+            If passed, it will be concatenated at end
+
+    Returns:
+        str
+            Sequence combined with whitespace.
+    """
+    if not input_element.endswith((" ", "\n", "\t")) and not output_element.startswith(
+        (" ", "\n", "\t")
+    ):
+        return input_element + " " + output_element + eos_token
+    return input_element + output_element + eos_token
+
+
+def get_data_collator(
+    packing: bool,
+    response_template: Optional[str],
+    tokenizer: AutoTokenizer,
+    is_traindata_tokenized: bool,
+    max_seq_length: int,
+) -> Callable:
+    """Create and return the the appropriate collator type based on the configuration for packing,
+    response_template, and dataset_text_field.
+
+    Args:
+        packing: bool
+            Whether or not we should apply packing or not.
+        response_template: Optional[str]
+            Response template to be used for formatting by TRL.
+        tokenizer: AutoTokenizer
+            Loaded tokenizer object to be used by the collator.
+        is_traindata_tokenized: bool
+            Whether train Dataset is tokenized or not
+        max_seq_length: int
+            Max sequence length expected
+
+    Returns:
+        Callable
+            Callable collator to be leveraged by the trainer.
+    """
+
+    if not packing:
+        # TODO: near term - how response template ids are parsed out needs to be cleaned.
+        # The [2:] here applies if response template has \n prefix, it is needed to strip \n,
+        # otherwise template is not found. We will create issue to clean this out after we discuss
+        # data formats and collators we will support.
+        if response_template:
+            response_template_ids = tokenizer.encode(
+                response_template, add_special_tokens=False
+            )[2:]
+            return DataCollatorForCompletionOnlyLM(
```

Comment:
so we only support these 2 `collators`? will we support anymore in the other code paths? this will impact the padding free plugin

Reply:
For now the data collators are not being touched by data preprocessor so this functionality is as it stands in the library currently. 
For future we will expand on collators as we grow use cases of data preprocessor.